---
title: "gwaslake: exploring the MRC GWAS API and data/software ecosystem with Bioconductor"
author: "Vincent J. Carey, stvjc at channing.harvard.edu"
date: "`r format(Sys.time(), '%B %d, %Y')`"
vignette: >
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteIndexEntry{gwaslake: exploring the MRC GWAS API and data/software ecosystem with Bioconductor}
  %\VignetteEncoding{UTF-8}
output:
  BiocStyle::html_document:
    highlight: pygments
    number_sections: yes
    theme: united
    toc: yes
---

# Introduction

We are indebted to the MRC Integrative Epidemiology Unit for
producing the infrastructure leading to the following schematic:

![mrc eco](mrcieuglue.png)

## Caveat about software component maintenance

The packages identified in the ecosystem are mostly
accessed through github repositories.  The interdependencies
and idiosyncrasies of using Remotes in DESCRIPTION made it
difficult to produce a container image, so I have forked several
of the packages identified in the schematic, to tweak the
DESCRIPTIONs.  I have filed some pull requests but have not
heard back.  I will try to keep the DESCRIPTION for this workshop
package current, minimizing use of forks, and hope individual
package maintainers will get in touch.

## Basic goals

- To understand the resources underlying the IEU GWAS API.
- To explore genetic analyses supported by the gwasglue
infrastructure.

# The API viewed through ieugwasr

We'll start by listing the functions available in the `ieugwasr` package.

```{r chk1}
suppressPackageStartupMessages({
library(gwaslake)
library(plotly)
library(ieugwasr)
library(dplyr)
library(DT)
})
ls("package:ieugwasr")
```

The pkgdown site for `ieugwasr` includes a reference listing, visible [here](https://mrcieu.github.io/ieugwasr/reference/index.html).

Let's try the gwasinfo command.  

```{r lkgwinf}
gwi = gwasinfo()
gwi
```

That's a huge table.  We can get an overview by tabulating the 'batches' into which
studies have been organized.

```{r lkbat}
gb = batches()
datatable(gb)
```


## Diseases available

A comprehensive view of phenotypes is challenging to assemble.
The `category` field has a relevant value, but we'll want to explore
further.
```{r lkdis}
 table(gwi$category)
distab = gwi %>% filter(category=="Disease")
distab %>% 
   group_by(trait) %>% summarise(n=n()) %>% arrange(desc(n))
```

A searchable table will be useful.
```{r lkdt}
datatable(as.data.frame(distab))
```

## A visual survey displaying study sizes for selected outcomes

The `survey_gwas` function plots number of cases versus number
of controls for all studies in which the `phrases` argument
matches the `trait` field, using `grep`.
With plotly,
we get an interactive graphic with additional data displayed on hover.
```{r lksurv}
ggplotly(survey_gwas(phrases=c("asthma", "asthmatic")))
```

## GWAS hits and their annotation

The [pkgdown document](https://mrcieu.github.io/ieugwasr/articles/guide.html) 
on `ieugwasr` is highly informative.  We work through some examples here
to verify functionality of our container.

First, by using the searchable table above, we can obtain the study
id for "Asthma": `ieu-a-44`.  We can then obtain the top "hits" for
this trait (a p-value threshold can be specified; default is 5e-8).
```{r lktop}
asthtop = tophits("ieu-a-44")
asthtop
```
Basic annotation on these hits can be obtained using
either positions or rsids.
```{r anno1}
asthv = variants_rsid(asthtop$rsid)
datatable(asthv)
```

## PheWAS/eQTL lookup

Let's follow up our hit in GSDMB with a 'PheWAS'.
The p-value threshold is indicated in documentation to be 0.00001.

```{r doph}
gdp = phewas("rs2290400")
datatable(gdp)
```

A number of the traits are genes, and the findings come from
eQTL studies.  Let's see where these genes are located.

```{r lkuppheq}
eqtab = gdp[grep("^eqtl", gdp$id),]
eqtab = bind_sym(eqtab)
table(eqtab$gchrom)
```

So there are 'trans' hits for our GSDMB-resident asthma SNP.


