<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>gwaslake workshop, Bioc 2021 • gwaslake</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="gwaslake workshop, Bioc 2021">
<meta property="og:description" content="gwaslake">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">gwaslake</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.0.20</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/gwaslake.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/bioc.html">bioc_ogw: connecting MRCIEU R packages to Bioconductor</a>
    </li>
    <li>
      <a href="../articles/workshop.html">gwaslake workshop, Bioc 2021</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/vjcitn/gwaslake">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="workshop_files/header-attrs-2.11/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>gwaslake workshop, Bioc 2021</h1>
                        <h4 class="author">Vincent Carey<a href="#fn1" class="footnote-ref" id="fnref1"><sup>1</sup></a>
</h4>
            
      
      <small class="dont-index">Source: <a href="https://github.com/vjcitn/gwaslake/blob/master/vignettes/workshop.Rmd"><code>vignettes/workshop.Rmd</code></a></small>
      <div class="hidden name"><code>workshop.Rmd</code></div>

    </div>

    
    
<div id="integrative-genetic-epidemiology-with-opengwas-opencravat-and-bioconductor" class="section level1">
<h1 class="hasAnchor">
<a href="#integrative-genetic-epidemiology-with-opengwas-opencravat-and-bioconductor" class="anchor"></a>Integrative genetic epidemiology with OpenGWAS, OpenCRAVAT, and Bioconductor</h1>
<p>Authors: Vincent Carey<a href="#fn2" class="footnote-ref" id="fnref2"><sup>2</sup></a>, <br> Last modified: 16 Mar, 2021.</p>
<div id="overview" class="section level2">
<h2 class="hasAnchor">
<a href="#overview" class="anchor"></a>Overview</h2>
<div id="description" class="section level3">
<h3 class="hasAnchor">
<a href="#description" class="anchor"></a>Description</h3>
<p>Along with the topic of your workshop, include how students can expect to spend their time. For the description may also include information about what type of workshop it is (e.g. instructor-led live demo, lab, lecture + lab, etc.). Instructors are strongly recommended to provide completely worked examples for lab sessions, and a set of stand-alone notes that can be read and understood outside of the workshop.</p>
</div>
<div id="pre-requisites" class="section level3">
<h3 class="hasAnchor">
<a href="#pre-requisites" class="anchor"></a>Pre-requisites</h3>
<p>List any workshop prerequisites, for example:</p>
<ul>
<li>Basic knowledge of R syntax</li>
<li>Familiarity with human SNPs</li>
</ul>
</div>
<div id="participation" class="section level3">
<h3 class="hasAnchor">
<a href="#participation" class="anchor"></a>Participation</h3>
<p>Students will use R to acquire variant data from resources like TCGA, select variant annotation resources available in OpenCRAVAT, and produce reports on variant function and impact.</p>
</div>
<div id="r-bioconductor-packages-used" class="section level3">
<h3 class="hasAnchor">
<a href="#r-bioconductor-packages-used" class="anchor"></a><em>R</em> / <em>Bioconductor</em> packages used</h3>
<p>GenomicRanges</p>
</div>
<div id="time-outline" class="section level3">
<h3 class="hasAnchor">
<a href="#time-outline" class="anchor"></a>Time outline</h3>
<table class="table">
<thead><tr class="header">
<th>Activity</th>
<th>Time</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>Concepts of genetic variation</td>
<td>10m</td>
</tr>
<tr class="even">
<td>MRC Integrative Epidemiology Unit resources</td>
<td>10m</td>
</tr>
<tr class="odd">
<td>OpenCRAVAT</td>
<td>10m</td>
</tr>
<tr class="even">
<td>Exercises</td>
<td>30m</td>
</tr>
</tbody>
</table>
</div>
<div id="workshop-goals-and-objectives" class="section level3">
<h3 class="hasAnchor">
<a href="#workshop-goals-and-objectives" class="anchor"></a>Workshop goals and objectives</h3>
<p>The interpretation of genetic variants is fundamental to all aspects of clinical genetics and genetic epidemiology. MRC OpenGWAS (publication <a href="https://doi.org/10.1101/2020.08.10.244293" class="uri">https://doi.org/10.1101/2020.08.10.244293</a>) is a data repository and API suite providing interactive access to statistics and metadata for hundreds of billions of human genetic variants. OpenCRAVAT (web site opencravat.org, publication DOI 10.1200/CCI.19.00132) is a system that amalgamates over 100 variant annotation resources and simplifies the development of rich characterizations of structural and functional contexts of genetic variants. Bioconductor (bioconductor.org) is an ecosystem of data structures and software packages that can be used in many contexts in genome biology and computational biomedicine. The gwaslake workshop adapts Bioconductor programming patterns and flexible containerization to simplify exploration, annotation, and interpretation of OpenGWAS variants assembled on a large collection of cohorts and phenotypes.</p>
</div>
<div id="learning-goals" class="section level3">
<h3 class="hasAnchor">
<a href="#learning-goals" class="anchor"></a>Learning goals</h3>
<p>In this workshop, we will guide you through the assembly of variants from diverse sources, in diverse formats, for flexible annotation using OpenCRAVAT. Bioconductor data structures and app designs are used to provide high-level conveniences for representation and analysis of cohorts arising in genetic epidemiology and cancer genomics.</p>
<ul>
<li>Import variation data from OpenGWAS and other human cohorts to R</li>
<li>Use Bioconductor tools and Rstudio apps to configure and execute annotation task processes defined by OpenCRAVAT</li>
<li>Retrieve annotation reports generated by OpenCRAVAT, moving the results forward in interactive workflows addressing epidemiologic and clinical interpretation</li>
</ul>
<p>Through live demonstrations and interactive small-group exercises, you will learn how to:</p>
</div>
<div id="learning-objectives" class="section level3">
<h3 class="hasAnchor">
<a href="#learning-objectives" class="anchor"></a>Learning objectives</h3>
<ul>
<li>Acquire data on human genetic variation for investigation with Bioconductor</li>
<li>Select annotation goals and tools from those available in OpenCRAVAT</li>
<li>Produce reports on functional and structural impacts of genetic variants using OpenCRAVAT in Bioconductor</li>
</ul>
<p>The activities undertaken in this workshop will require some familiarity with either jupyter notebooks or Rstudio, but no programming per se will be necessary to take advantage of the workshop material.</p>
<p>Learning Objectives:</p>
<ol style="list-style-type: decimal">
<li><p>Identify the fundamental components of working with human genetic variants in a cloud-native environment, making use of diverse GWAS and PheWAS results</p></li>
<li><p>Understand how to define a series of annotation tasks for variant data assembled on individuals or cohorts, and how to use OpenCRAVAT to execute these tasks with a Bioconductor/shiny app</p></li>
<li><p>Use the outputs of OpenCRAVAT annotators to interpret effects of genetic variants on disease risk or other concepts of interest in genetic epidemiology.</p></li>
</ol>
</div>
</div>
</div>
<div class="footnotes">
<hr>
<ol>
<li id="fn1"><p><a href="mailto:stvjc@channing.harvard.edu" class="email">stvjc@channing.harvard.edu</a><a href="#fnref1" class="footnote-back">↩︎</a></p></li>
<li id="fn2"><p>Harvard Medical School<a href="#fnref2" class="footnote-back">↩︎</a></p></li>
</ol>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Vince Carey.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
